internal/operands/common-instancetypes/reconcile.go

```diff
@@ -58,52 +98,196 @@ func fetchClusterResources[C clusterType](bundlePath string) ([]C, error) {
 	}
 }
 
-func New(instancetypeBundlePath, preferenceBundlePath string) (operands.Operand, error) {
-	virtualMachineClusterInstancetypes, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterInstancetype](instancetypeBundlePath)
+func FetchBundleResource[C clusterType](path string) ([]C, error) {
+	file, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	virtualMachineClusterPreferences, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterPreference](preferenceBundlePath)
+	return decodeResources[C](file)
+}
+
+func (c *commonInstancetypes) fetchResourcesFromBundle() ([]instancetypev1alpha2.VirtualMachineClusterInstancetype, []instancetypev1alpha2.VirtualMachineClusterPreference, error) {
+	virtualMachineClusterInstancetypes, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterInstancetype](c.virtualMachineClusterInstancetypeBundle)
+	if err != nil {
+		return nil, nil, err
+	}
+	virtualMachineClusterPreferences, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterPreference](c.virtualMachineClusterPreferenceBundle)
+	if err != nil {
+		return nil, nil, err
+	}
+	return virtualMachineClusterInstancetypes, virtualMachineClusterPreferences, err
+}
+
+func generateKustomization(URL string) (filesys.FileSystem, string, error) {
+	fSys := filesys.MakeFsOnDisk()
+	tmpDir, err := filesys.NewTmpConfirmedDir()
+	if err != nil {
+		return nil, "", err
+	}
+	if err = fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(fmt.Sprintf("\nresources:\n  - %s", URL))); err != nil {
+		return nil, "", err
+	}
+	return fSys, tmpDir.String(), nil
+}
+
+func generateResourcesFromURL(URL string) (resmap.ResMap, error) {
+	fSys, tmpDirPath, err := generateKustomization(URL)
 	if err != nil {
 		return nil, err
 	}
-	return &commonInstancetypes{
-		virtualMachineClusterInstancetypes: virtualMachineClusterInstancetypes,
-		virtualMachineClusterPreferences:   virtualMachineClusterPreferences,
-	}, nil
+	k := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+	return k.Run(fSys, tmpDirPath)
 }
 
-func (c *commonInstancetypes) Name() string {
-	return operandName
+func decodeResMapResources[C clusterType](r *resource.Resource) ([]C, error) {
+	bytes, err := r.AsYAML()
```

Comment:
Can you encode it to json, instead of yaml? The `AsYAML()` method first encodes it to json and then translates json to yaml. We can skip the translation step, because the `decodeResources()` below can load json.